src/StationAPIConnector.php

```diff
@@ -0,0 +1,304 @@
+use GuzzleHttp\RequestOptions;
+use Drupal\Core\File\FileSystemInterface;
+
+/**
+ * Connects to station locator API.
+ */
+class StationAPIConnector {
+
+  /**
+   * Drupal\Core\Config\ConfigFactoryInterface definition.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Drupal\Core\Logger\LoggerChannelFactory definition.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $logger;
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * Constructor for the class.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Config factory.
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $logger
+   *   Logger.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   File system.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, LoggerChannelFactory $logger, FileSystemInterface $file_system) {
+    $this->configFactory = $configFactory;
+    $this->logger = $logger;
+    $this->fileSystem = $file_system;
+  }
+
+  /**
+   * Helper method to connect API.
+   *
+   * @return mixed
+   *   Client Id, Access token, Rdm auth values.
+   */
+  public function connectApi() {
+    try {
+      // Get the folder.
+      $folder = (getenv('LAGOON_ENVIRONMENT_TYPE') !== 'production') ? 'nonprod' : 'production';
+      // Path to your .crt and .key files.
+      $certPath = '/app/keys/' . $folder . '/api.crt';
+      $keyPath = '/app/keys/' . $folder . '/api.pem';
+      // Path to the CA bundle file.
+      $caBundlePath = '/app/keys/cacert.pem';
+
+      // Initiating variables to escape phpcs errors.
+      $apiUrl = getenv('API_URL');
+
+      // OAuth configuration variables.
+      $tokenEndpoint = getenv('TOKEN_END_POINT');
+      $clientId = getenv('CLIENT_ID');
+      $clientSecret = getenv('CLIENT_SECRET');
+      $rdm_auth = getenv('RDM_AUTH');
+
+      $client = new Client([
+        'base_uri' => $apiUrl,
+        'curl' => [
+          CURLOPT_SSLCERT => $certPath,
+          CURLOPT_SSLKEY => $keyPath,
+          CURLOPT_CAINFO => $caBundlePath,
+        ],
+      ]);
+
+      $accessToken = NULL;
+
+      $response = $client->post($tokenEndpoint, [
+        RequestOptions::FORM_PARAMS => [
+          'grant_type' => 'client_credentials',
+          'client_id' => $clientId,
+          'client_secret' => $clientSecret,
+        ],
+      ]);
+
+      $data = json_decode($response->getBody()->getContents(), TRUE);
+      $accessToken = $data['access_token'];
+
+      return [
+        'clientId' => $client,
+        'accessToken' => $accessToken,
+        'rdm_auth' => $rdm_auth,
+      ];
+    }
+    catch (BadResponseException $e) {
+      $this->logger->get('tide_station_locator connect API.')
+        ->error('BadResponse error: @error', ['@error' => $e->getResponse()]);
```

Comment:
@g2410g If there is an error in the call, we are getting this error:
`Error: Object of class GuzzleHttp\Psr7\Response could not be converted to string in`